service/src/main/java/uk/gov/hmcts/reform/fpl/utils/HearingDateHelper.java

```diff
@@ -0,0 +1,31 @@
+package uk.gov.hmcts.reform.fpl.utils;
+
+import uk.gov.hmcts.reform.fpl.model.common.dynamic.DynamicElementIndicator;
+import uk.gov.hmcts.reform.fpl.model.common.dynamic.DynamicListElement;
+import uk.gov.hmcts.reform.fpl.service.DateFormatterService;
+
+import java.time.LocalDate;
+import java.time.format.FormatStyle;
+import java.util.UUID;
+
+public class HearingDateHelper implements DynamicElementIndicator {
+    private LocalDate date;
+    private UUID id;
+    private final DateFormatterService dateFormatterService;
+
+    public HearingDateHelper(UUID id, LocalDate date, DateFormatterService dateFormatterService) {
+        this.id = id;
+        this.date = date;
+        this.dateFormatterService = dateFormatterService;
+    }
+
+    @Override
+    public DynamicListElement toDynamicElement() {
+        final String dateString = convertDate(date);
+        return DynamicListElement.builder().code(id.toString()).label(dateString).build();
```

Comment:
why is the code not a UUID?

Reply:
I think it was when I initially made the DynamicListElement class and was trying to conform perfectly with [this](https://tools.hmcts.net/confluence/pages/viewpage.action?pageId=930742860) if you think that it would be fine as an object then it can be changed

Reply:
Discussed with @swalker125, using a UUID in this case should be fine

Reply:
updated to UUID